core/src/main/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryParser.java

```diff
@@ -221,7 +221,7 @@ private String parseFiltersAndFunctions(QueryParseContext parseContext, XContent
                         functionWeight = parser.floatValue();
                     } else {
                         if ("filter".equals(currentFieldName)) {
-                            filter = parseContext.parseInnerFilter();
+                            filter = parseContext.context().parseInnerFilter();
```

Comment:
I am not sure why we moved these parseInnerFilter methods to QueryCreationContext. Didn't they make more sense in QueryParseContext (although they will be removed at some point)?